runtime/gc_glue_java/ArrayletObjectModelBase.hpp

```diff
@@ -131,23 +152,24 @@ class GC_ArrayletObjectModelBase
 	{
 		bool needAlignment = false;
 
-#if defined(OMR_GC_COMPRESSED_POINTERS)
-		/* Compressed pointers require that each leaf starts at 8 aligned address.
-		 * Otherwise compressed leaf pointers will not work with shift value of 3.
-		 */
-		needAlignment = true;
-#else
-		/* The alignment padding is only required when the size of the spine pointers are
-		 * not 8 byte aligned.  For example, on a 64-bit non-compressed platform, a single
-		 * spine pointer would be 8 byte aligned, whereas on a 32-bit platform, a single
-		 * spine pointer would not be 8 byte aligned, and would require additional padding.
-		 */
-		if (sizeof(fj9object_t) < sizeof(double)) {
-			if (OBJECT_HEADER_SHAPE_DOUBLES == J9GC_CLASS_SHAPE(clazz)) {
-				needAlignment = true;
+		if (compressObjectReferences()) {
+			/* Compressed pointers require that each leaf starts at 8 aligned address.
+			 * Otherwise compressed leaf pointers will not work with shift value of 3.
+			 */
```

Comment:
in case of 4-bit shift an alignment would be 16. So I suggest to make this comment more common.

Reply:
Comment updated.